src/metabolic_ninja/tasks.py

```diff
@@ -333,17 +338,57 @@ def evaluate_cofactor_swap(designs, pathway, model, method):
     if designs is None:
         return []
     logger.info(f"Evaluating {len(designs)} co-factor swap designs.")
+    source_pair = ("nad_c", "nadh_c")
+    target_pair = ("nadp_c", "nadph_c")
     results = []
-    for row in designs.data_frame.itertuples(index=False):
+    # FIXME (Moritz Beber): The model context is currently bugged.
+    #  See https://github.com/opencobra/cobrapy/issues/849
+    #  We need to make a copy.
+    model = model.copy()
+    source_a = model.metabolites.get_by_id(source_pair[0])
+    source_b = model.metabolites.get_by_id(source_pair[1])
+    target_a = model.metabolites.get_by_id(target_pair[0])
+    target_b = model.metabolites.get_by_id(target_pair[1])
+    for design in designs.data_frame.itertuples(index=False):
+        manipulations = []
+        for rxn_id in design.targets:
+            rxn: Reaction = model.reactions.get_by_id(rxn_id)
+            metabolites = rxn.metabolites
+            # Swap from source to target co-factors.
+            if source_a in metabolites:
+                metabolites[target_a] = metabolites[source_a]
+                metabolites[target_b] = metabolites[source_b]
+                metabolites[source_a] = 0
+                metabolites[source_b] = 0
+                manipulations.append({"id": rxn_id, "from": [source_pair[0], source_pair[1]], "to": [target_pair[0], target_pair[1]]})
+            elif target_a in metabolites:
+                metabolites[source_a] = metabolites[target_a]
+                metabolites[source_b] = metabolites[target_b]
+                metabolites[target_a] = 0
+                metabolites[target_b] = 0
+                manipulations.append({"id": rxn_id, "from": [target_pair[0], target_pair[1]], "to": [source_pair[0], source_pair[1]]})
+            else:
+                raise KeyError(
+                    f"Neither co-factor swap partner present in "
+                    f"predicted target reaction '{rxn_id}'."
+                )
+            rxn.add_metabolites(metabolites, combine=False)
+        logger.info("Calculating production values.")
+        with model:
+            production = evaluate_production(model, pathway.product.id, model.carbon_source)
+        logger.info("Calculating biomass coupled production values.")
+        with model:
+            biomass_coupled_production = evaluate_biomass_coupled_production(
```

Comment:
If you are only interested in a few values, Python has a neat new syntax:

```suggestion
            growth, *_ = evaluate_biomass_coupled_production(
```